pkg/controller/controller_utils.go

```diff
@@ -825,9 +832,8 @@ func (s ActivePodsWithRanks) Less(i, j int) bool {
 		return len(s.Pods[i].Spec.NodeName) == 0
 	}
 	// 2. PodPending < PodUnknown < PodRunning
-	m := map[v1.PodPhase]int{v1.PodPending: 0, v1.PodUnknown: 1, v1.PodRunning: 2}
-	if m[s.Pods[i].Status.Phase] != m[s.Pods[j].Status.Phase] {
-		return m[s.Pods[i].Status.Phase] < m[s.Pods[j].Status.Phase]
+	if podPhaseToOrdinal[s.Pods[i].Status.Phase] != podPhaseToOrdinal[s.Pods[j].Status.Phase] {
```

Comment:
Again, if `s[i].Status.Phase != s[j].Status.Phase` is valid above, it should be fine here too.